go-controller/pkg/node/node.go

```diff
@@ -571,10 +573,48 @@ func (n *OvnNode) Start(ctx context.Context, wg *sync.WaitGroup) error {
 		}
 	}
 
+	// Start the health checking server used by egressip, if EgressIPNodeHealthCheckPort is specified
+	if err := n.startEgressIPHealthCheckingServer(wg, mgmtPortConfig); err != nil {
+		return err
+	}
+
 	klog.Infof("OVN Kube Node initialized and ready.")
 	return nil
 }
 
+func (n *OvnNode) startEgressIPHealthCheckingServer(wg *sync.WaitGroup, mgmtPortConfig *managementPortConfig) error {
+	healthCheckPort := config.OVNKubernetesFeature.EgressIPNodeHealthCheckPort
+	if healthCheckPort == 0 {
+		klog.Infof("Egress IP health check server skipped: no port specified")
+		return nil
+	}
+
+	var nodeMgmtIP net.IP
+	if mgmtPortConfig.ipv4 != nil {
+		nodeMgmtIP = mgmtPortConfig.ipv4.ifAddr.IP
+	} else if mgmtPortConfig.ipv6 != nil {
+		nodeMgmtIP = mgmtPortConfig.ipv6.ifAddr.IP
+		// Wait for IPv6 address to become usable.
```

Comment:
@trozet this is the settle call we need in order to support cases when ipv6 is the only stack available.